src/ebmc/k_induction.cpp

```diff
@@ -90,14 +91,20 @@ int k_inductiont::operator()()
   if(result != -1)
     return result;
 
+  if(properties.properties.empty())
+  {
+    message.error() << "no properties" << messaget::eom;
+    return 1;
+  }
+
   // liveness to safety translation, if requested
   if(cmdline.isset("liveness-to-safety"))
     liveness_to_safety(transition_system, properties);
 
-  if(properties.properties.empty())
+  // Check whether the properties are suitable for k-induction.
+  if(properties.requires_lasso_constraints())
   {
-    message.error() << "no properties" << messaget::eom;
-    return 1;
+    throw ebmc_errort() << "k-induction does not support liveness properties";
```

Comment:
I think it would be useful if this error included a source location (e.g., the source location of the first liveness property that we find. This, of course, requires adjusting the `requires_lasso_constraints` interface so might be a change for a future PR, but would still make it easier for a user.

Reply:
Now with source location